3 services

SeverinBuchser opened the pull request:
I made many file changes, but most of them are small scss changes: I implemented a better import for the variables in scss, which allows for a `@import 'variables'`. 

The rest of the changes are in the services directory concerning mostly the user and login user service.

SeverinBuchser committed:
Experimental Branch: New Approach for user login and handling in general

SeverinBuchser committed:
Fix Services and improve design

The services now use a new design following the observer pattern with an extension. The extension is loading. A service can now load a source and then notify the observer when starting, loading and finishing. The observer then also receives either an error or the value loaded. 

The other fixes were fixes with scss implementations.

medoparder commented:
Reviwed and most is ok. 
- by the rent-dialog the rent-button has no css yet, is gray. 
Login and register everything is ok. services seems to be ok.

SeverinBuchser committed:
Improve Card Components

The cards components like the login, register etc are now setup in a more generic way, which allows the components to be used in a non centered way, for example on the user profile. This is not implemented yet.

The implementation is done with a card component (CenterCardComponent), which just takes ng content, title and error message as an input.

SeverinBuchser committed:
Refactor imports in backend

SeverinBuchser commented:
I fixed the rent-dialog as well as improved some other things.
